CMakeLists.txt

```diff
@@ -253,6 +253,10 @@ if(TRITON_BUILD_PYTHON_MODULE)
   target_link_options(triton PRIVATE ${LLVM_LDFLAGS})
 endif()
 
+if (UNIX AND NOT APPLE)
```

Comment:
What does not APPLE mean here?

Reply:
This PR didn't handle the case in MacOS, but there is a possible way to use exported_symbols_lists to export a whitelist symbols as [mxnet/CMakeLists.txt](https://github.com/apache/mxnet/blob/b84609d3fc73d20929c114eab95faaa56e6c5ede/CMakeLists.txt#L740).

Reply:
This PR didn't handle the case in MacOS, but there is a possible way to use exported_symbols_lists to export a whitelist symbols as [mxnet/CMakeLists.txt](https://github.com/apache/mxnet/blob/b84609d3fc73d20929c114eab95faaa56e6c5ede/CMakeLists.txt#L740).

Reply:
We want to add the linker option only on unix and not MacOS.